src/cli/global/install.rs

```diff
@@ -212,13 +216,23 @@ pub(super) async fn create_executable_scripts(
     prefix: &Prefix,
     shell: &ShellEnum,
     activation_script: String,
+    no_activation: bool,
 ) -> miette::Result<()> {
     for BinScriptMapping {
         original_executable: exec,
         global_binary_path: executable_script_path,
     } in mapped_executables
     {
-        let mut script = activation_script.clone();
+        let mut script = if cfg!(unix) {
+            "#!/bin/sh\n".to_string()
+        } else {
+            String::new()
+        };
+
+        if !no_activation {
+            script.push_str(&activation_script);
+        }
```

Comment:
```suggestion
        let mut script = if no_activation {
            if cfg!(unix) {
                "#!/bin/sh\n".to_string()
            } else {
                String::new()
            }
        } else {
            activation_script.clone()
        }
```

Otherwise, you'll get `#!/bin/sh` twice if `--no-activation` isn't passed